### DISCUSSION TITLE
Thread safe?

### DISCUSSION THREAD

**Discussion by pmarflee** [ctolkien/Slugify#152] [Q&A] [answered]
Is SlugHelper thread-safe? It looks like it is, as long as you don't use `Config.DeniedCharactersRegex`.

I want to use a singleton instance of SlugHelper in my application.

**Comment by ctolkien** [comment, marked as answer]
Looping back around on this. This lib is not _technically_ guaranteed to be thread safe, however in practice in the "regular" use-case, it should be in practice. Unless you're doing things like altering configuration on the fly of a shared configuration object - that will likely cause inconsistent behaviour, but it's also something I don't see people actually doing in practice.

In the new V5, rather than passing in a string for the `DeniedCaractersRegex`, you'll pass in a `Regex` object - which are thread safe.

**Comment by ctolkien** [comment]
Going to move this to discussions.
